src/utils/engine.ts

```diff
@@ -1,37 +1,66 @@
+import { getConfig } from '../commands/config';
+import { AnthropicEngine } from '../engine/anthropic';
+import { AzureEngine } from '../engine/azure';
 import { AiEngine } from '../engine/Engine';
-import { OpenAi } from '../engine/openAi';
+import { FlowiseAi } from '../engine/flowise';
 import { Gemini } from '../engine/gemini';
-import { getConfig } from '../commands/config';
 import { OllamaAi } from '../engine/ollama';
-import { AnthropicAi } from '../engine/anthropic'
-import { TestAi } from '../engine/testAi';
-import { Azure } from '../engine/azure';
-import { FlowiseAi } from '../engine/flowise'
+import { OpenAiEngine } from '../engine/openAi';
+import { TestAi, TestMockType } from '../engine/testAi';
 
 export function getEngine(): AiEngine {
   const config = getConfig();
-  const provider = config?.OCO_AI_PROVIDER;
-  
-  if (provider?.startsWith('ollama')) {
-    const ollamaAi = new OllamaAi();
-    const model = provider.substring('ollama/'.length);
-    if (model) {
-      ollamaAi.setModel(model);
-      ollamaAi.setUrl(config?.OCO_OLLAMA_API_URL);
-    }
-    return ollamaAi;
-  } else if (provider == 'anthropic') {
-    return new AnthropicAi();
-  } else if (provider == 'test') {
-    return new TestAi();
-  } else if (provider == 'gemini') {
-    return new Gemini();  
-  } else if (provider == 'azure') {
-  	return new Azure();
-  } else if( provider == 'flowise'){
-    return new FlowiseAi();
+  const provider = config.OCO_AI_PROVIDER;
+
+  const DEFAULT_CONFIG = {
+    model: config.OCO_MODEL!,
+    maxTokensOutput: config.OCO_TOKENS_MAX_OUTPUT!,
+    maxTokensInput: config.OCO_TOKENS_MAX_INPUT!,
+    baseURL: config.OCO_OPENAI_BASE_PATH!
+  };
+
+  switch (provider) {
+    case provider?.startsWith('ollama') && provider:
+      const model = provider.substring('ollama/'.length);
+      return new OllamaAi({
+        ...DEFAULT_CONFIG,
+        apiKey: '',
+        model,
+        baseURL: config.OCO_OLLAMA_API_URL!
+      });
+
+    case 'anthropic':
+      return new AnthropicEngine({
+        ...DEFAULT_CONFIG,
+        apiKey: config.OCO_ANTHROPIC_API_KEY!
+      });
+
+    case 'test':
+      return new TestAi(config.OCO_TEST_MOCK_TYPE as TestMockType);
+
+    case 'gemini':
+      return new Gemini({
+        apiKey: config.OCO_GEMINI_API_KEY!,
+        ...DEFAULT_CONFIG
+      });
+
+    case 'azure':
+      return new AzureEngine({
+        apiKey: config.OCO_AZURE_API_KEY!,
+        ...DEFAULT_CONFIG
+      });
+
+    case 'flowise':
+      return new FlowiseAi({
+        ...DEFAULT_CONFIG,
+        baseURL: config.OCO_FLOWISE_ENDPOINT || DEFAULT_CONFIG.baseURL,
```

Comment:
OCO_FLOWISE_ENDPOINT will be deprecated together with OCO_OPENAI_BASE_PATH and OCO_GEMINI_BASE_PATH

all of the above will be one variable: OCO_API_URL

in this specific case it's expected to include full flowise url with `/api/v1/prediction`

Reply:
changing the usage also

[before](https://github.com/di-sukharev/opencommit/pull/391/files#diff-c8b1e3bdb0481b57f7f45af3c817342689c81ce3097dad6729bd643c7dedef50L22%20) [after](https://github.com/di-sukharev/opencommit/pull/391/files#diff-c8b1e3bdb0481b57f7f45af3c817342689c81ce3097dad6729bd643c7dedef50R14)